### PR TITLE
EES-3607 find stats sorting

### PR DESCRIPTION
**Pull request by amyb-hiveit**
Adds sorting to the new Find Statistics page. The existing Find Statistics page should not be affected, to see the new one use the newDesign=true query parameter, e.g. http://localhost:3000/find-statistics?newDesign=true

- doesn't actually do any sorting yet as using dummy data
- displays as a select dropdown on mobile and a radio group on desktop
- I've added the mobile filters button just to sort out the styling for that section, it doesn't do anything yet.
- includes a non-javascript only submit button to the controls so you can submit the form and sort the results when js is disabled. 
- pagination gets reset when results are sorted
- a Google Analytics event is fired when change the sorting
- I've added an inline option to FormSelect for use on the mobile version

sorting-mobile
sorting-desktop

### CODE REVIEW

File: src/explore-education-statistics-frontend/src/modules/find-statistics/FindStatisticsPageNew.tsx [hunk 1]
```diff
@@ -1,29 +1,95 @@
...
+  >(publications);
+  const [currentSortBy, setCurrentSortBy] = useState<
+    PublicationSortOptions | undefined
+  >(sortBy ?? undefined);
+  const [isLoadingPublications, toggleIsLoadingPublications] = useToggle(false);
+
+  useEffect(() => {
+    setCurrentPage(page);
+    setCurrentPublications(publications);
+  }, [page, publications]);
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/FindStatisticsPageNew.tsx:51]: How come we're doing everything in state again?

Noticed your comment that there are 'usability problems as the page reloads while you're interacting with the sort controls' . What actually happens?

I stripped out all the state and changed the `router.push` back to be non-shallow and everything seems to work fine. Focus is not lost and I can cycle through the sort options easily enough. Is there a screen reader issue here?

File: src/explore-education-statistics-frontend/src/modules/find-statistics/components/SortControls.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  { label: 'Oldest', value: 'oldest' },
+  { label: 'A to Z', value: 'alphabetical' },
+];
+
+interface FormValues {
+  sortBy: PublicationSortOptions;
+}
+
+interface Props {
+  initialValues: FormValues;
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/components/SortControls.tsx:19]: Don't need to do this as `initialValues` as this isn't a particularly complicated form (we're just concerned with sorting). Could just use a `sortBy` prop to simplify things.

File: src/explore-education-statistics-frontend/src/modules/find-statistics/components/__tests__/SortControls.test.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    expect(sortOptions[0]).toEqual(sortGroup.getByLabelText('Newest'));
+    expect(sortOptions[0]).toBeChecked();
+    expect(sortOptions[1]).toEqual(sortGroup.getByLabelText('Oldest'));
+    expect(sortOptions[1]).not.toBeChecked();
+    expect(sortOptions[2]).toEqual(sortGroup.getByLabelText('A to Z'));
+    expect(sortOptions[2]).not.toBeChecked();
+
+    expect(screen.queryByLabelText('Sort results')).not.toBeInTheDocument();
+  });
+
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/components/__tests__/SortControls.test.tsx:35]: We should have a test to check that providing a different initial `sortBy` sets a different selected value (for desktop and mobile).

File: src/explore-education-statistics-common/src/components/form/FormSelect.tsx [hunk 4]
```diff
@@ -57,69 +60,104 @@ const FormSelect = ({
 }: FormSelectProps) => {
   return (
     <>
-      <FormLabel id={id} label={label} hideLabel={hideLabel} />
+      <FormSelectWrapper inline={inline}>
```
Comment (ntsim) [inline src/explore-education-statistics-common/src/components/form/FormSelect.tsx:63]: Don't need to split out a separate component for this. It's a bit of unnecessary indirection.

Would just implement this as a wrapping `div` with conditional `className` for simplicity.

**Review by ntsim — CHANGES_REQUESTED** [review]

File: src/explore-education-statistics-frontend/src/modules/find-statistics/FindStatisticsPageNew.tsx (hunk 1 above)
Comment (amyb-hiveit) [reply src/explore-education-statistics-frontend/src/modules/find-statistics/FindStatisticsPageNew.tsx:51]: I'm not sure what I was doing before to make it not work properly! I've taken it back out again and all works fine.

File: src/explore-education-statistics-frontend/src/modules/find-statistics/FindStatisticsPageNew.tsx [hunk 5]
```diff
@@ -83,41 +149,53 @@ const FindStatisticsPageNew: NextPage<Props> = ({ paging, publications }) => {
...
+                  variant="secondary"
+                >
+                  Filter results
+                </Button>
+              )}
+              <SortControls
+                initialValues={{ sortBy: currentSortBy ?? 'newest' }}
+                onChange={sortPublications}
+              />
+            </div>
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-frontend/src/modules/find-statistics/FindStatisticsPageNew.tsx]: Showed this to Marv and he agreed so have changed it as you suggest.

File: src/explore-education-statistics-common/src/components/form/FormSelect.tsx (hunk 4 above)
Comment (amyb-hiveit) [reply src/explore-education-statistics-common/src/components/form/FormSelect.tsx:63]: Unnecessary divs make me sad.

**Review by ntsim — APPROVED** [review]

**Pull request merged**